apisix/plugins/log-rotate.lua

```diff
@@ -315,6 +319,11 @@ end
 
 
 function _M.init()
+    if not local_conf then
+        local_conf = core.config.local_conf()
+    end
+    local nginx_config = local_conf and local_conf.nginx_config
```

Comment:
you can use core.table.try_read_attr() function instead. It's much cleaner.

Reply:
Nice function! I will try it.